File: Online Judges/LeetCode/July LeetCoding Challenge/Week-1/Plus One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if digits == []:
            return [1]
        if digits[-1] == 9:
            digits = self.plusOne(digits[:-1])
            digits.extend([0])
        else:
            digits[-1] += 1

        return digits

File: Online Judges/LeetCode/July LeetCoding Challenge/Week-1/Plus One.rs
impl Solution {
    pub fn plus_one(mut digits: Vec<i32>) -> Vec<i32> {
        for digit in digits.iter_mut().rev() {
            *digit += 1;
            if *digit == 10 {
                *digit = 0;
            }
            else {
                return digits;
            }
        }
        digits.insert(0,1);
        return digits;
    }
}